src/helper/ssg/index.ts

```diff
@@ -0,0 +1,74 @@
+import * as path from 'path'
+import { inspectRoutes } from '../../helper/dev'
+import type { Hono } from '../../hono'
+import type { Env, Schema } from '../../types'
+
+export interface FileSystemModule {
+  writeFile(path: string, data: string | Buffer): Promise<void>
+  mkdir(path: string, options: { recursive: boolean }): Promise<void | string>
+}
+
+export interface ToSSGResult {
+  success: boolean
+  files?: string[]
+  error?: Error
+}
+
+const generateFilePath = (routePath: string, outDir: string) => {
+  const fileName = routePath === '/' ? 'index.html' : routePath + '.html'
+  return path.join(outDir, fileName)
+}
+
+export const generateHtmlMap = async <
+  E extends Env = Env,
+  S extends Schema = {},
+  BasePath extends string = '/'
+>(
+  app: Hono<E, S, BasePath>
+): Promise<Map<string, string>> => {
+  const htmlMap = new Map<string, string>()
+  const baseURL = 'http://localhost'
+
+  for (const route of inspectRoutes(app)) {
+    if (route.isMiddleware) continue
+
+    const url = new URL(route.path, baseURL).toString()
+    const response = await app.fetch(new Request(url))
+    const html = await response.text()
+
+    htmlMap.set(route.path, html)
+  }
+
+  return htmlMap
+}
+
+export const saveHtmlToLocal = async (
+  htmlMap: Map<string, string>,
+  fsModule: FileSystemModule,
+  outDir: string
+): Promise<string[]> => {
+  const files: string[] = []
+
+  for (const [routePath, html] of htmlMap) {
+    const filePath = generateFilePath(routePath, outDir)
+    const dirPath = path.dirname(filePath)
+
+    await fsModule.mkdir(dirPath, { recursive: true })
+    await fsModule.writeFile(filePath, html)
+    files.push(filePath)
+  }
+
+  return files
+}
+
```

Comment:
I think `helper/ssg` should implement `toSSG` not only exporting `ToSSGInterface`.

```ts
export const toSSG: ToSSGInterface = async (app, fs, options) => {
  try {
    const maps = await generateHtmlMap(app)
    const files = await saveHtmlToLocal(maps, fs, options.dir)
    return { success: true, files }
  } catch (error) {
    const errorObj = error instanceof Error ? error : new Error(String(error))
    return { success: false, error: errorObj }
  }
}
```

Then, for Node.js, you can do the following without an adapter. Also, for platforms other than Node.js, Bun, and Deno, a user can just import `toSSG()` from `hono/ssg`.

```ts
import { toSSG } from 'hono/ssg'
import fs from 'node:fs/promises'

toSSG(app, fs, { dir: './static '})
```

Also, for example, Bun's adapter can be written as follows:

```ts
import { toSSG as baseToSSG } from '../../helper/ssg'

// ...

export const toSSG: ToSSGInterface = async (app, bunFileSystemModule, options: { dir: string }) => {
  return baseToSSG(app, bunFileSystemModule, options)
}
```

Reply:
The toSSG in deno and bun's Adaptor is now just a wrapper, so we made an interface that omits the fs, at least for convenience. If you don't need this realization, you probably don't need to put toSSG in Adaptor in the first place.